### ISSUE TITLE
Cannot install ruby 2.3 and olders with rbenv.

### ISSUE THREAD

**Issue by khuongduybui** [WhitewaterFoundry/WLinux#135]
I can use rbenv to install 2.5.x and 2.4.x, but 2.3 and older versions cannot be installed with rbenv.

I got errors such as:
https://pastebin.com/agdE5Kr6 (2.3.8)

**Comment by sirredbeard** [comment]
Are you still having issues? 

Have you by chance checked the rbenv issue page?

**Comment by khuongduybui** [comment]
1. Yes I still have the issue.
2. No I did not check rbenv
3. Unfortunately I updated to build 18272 so WSL is not working on my test machine. I will wait for next build and try again.

**Comment by LemonAndroid** [comment]
@khuongduybui ping me for a Zoom or TeamViewer session at manu@korfmann.info. cc @sirredbeard

**Comment by khuongduybui** [comment]
Sure, thanks. Just as soon as WSL starts working again.

**Comment by sirredbeard** [comment]
18277 went out today. Please give it a try and report back.

**Comment by khuongduybui** [comment]
I installed a fresh setup from the Store, then ran wlinux-setup and only selected to install ruby.
2.5.3 seemed to install just fine.

```sh
...
Installed ruby-2.5.3 to ~/.rbenv/versions/2.5.3
/tmp/tmp.2VnJxrljqZ
Checking ruby version
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
Installing bundler using gem
Fetching: bundler-1.17.1.gem (100%)
Successfully installed bundler-1.17.1
Parsing documentation for bundler-1.17.1
Installing ri documentation for bundler-1.17.1
Done installing documentation for bundler after 10 seconds
1 gem installed
Rehashing rbenv
Returning to ~
Cleaning up /tmp/tmp.2VnJxrljqZ
Skipping RAILS
```

Then I ran rbenv install 2.3.8 and got https://pastebin.com/vWwX7Aw2.

**Comment by sirredbeard** [comment]
It looks like a build issue with Ruby, would you agree?

What is the output of:
`$ apt-cache policy build-essential` and
`$ apt-cache policy gcc`?

I may need to go try to duplicate this issue this weekend.

**Comment by khuongduybui** [comment]
Yes it seems. I tried with both ruby-build and ruby-install and both failed.
```
build-essential:
    Installed: 12.5
    Candidate: 12.5
Version table:
*** 12.5 501
        501 https://deb.debian.org/debian testing/main amd64 Packages
        100 /var/lib/dpkg/status
    12.3 990
        990 https://deb.debian.org/debian stable/main amd64 Packages

gcc:
    Installed: 4:8.2.0-2
    Candidate: 4:8.2.0-2
Version table:
*** 4:8.2.0-2 501
        501 https://deb.debian.org/debian testing/main amd64 Packages
        100 /var/lib/dpkg/status
    4:6.3.0-4 990
        990 https://deb.debian.org/debian stable/main amd64 Packages
```

**Comment by khuongduybui** [comment]
For what it's worth, I installed `rbenv` on a fresh Ubuntu (18.04) WSL instance (using https://github.com/rbenv/rbenv-installer).

Trying `rbenv install 2.3.8` failed with the suggestion of installing `libssl-dev`, `libreadline-dev`, and `zlib1g-dev`.

I installed these packages with `apt` and tried again, it still failed but with the suggestion of installing `libssl-dev` only this time (which as evidently already installed).

Finally I figured it out and installed `libssl1.0-dev` with `apt` instead (which replaced `libssl-dev`). Then I managed to install `2.3.8` successfully. Will try to replicate the same "solution" to WLinux,

**Comment by khuongduybui** [comment]
After installing libssl1.0-dev I can install ruby 2.3.8 on WLinux too.

Not sure if we should add that to wlinux-setup script or let users figure it out ❓

**Comment by sirredbeard** [comment]
Glad you were able to get that worked out.

Added to wlinux-setup in #176.

Let me know if you run into future issues.

Thank you for your help and your patience.

Hayden

**Comment by khuongduybui** [comment]
1. The current script doesn't work because of conflicts. To install `libssl1.0` we need to remove `libssl-dev` and 
`libcurl4-openssl-dev`.

```
The following packages have unmet dependencies:                                                                                                                 libcurl4-openssl-dev : Conflicts: libssl1.0-dev but 1.0.2o-1 is to be installed                                                                                libssl-dev : Conflicts: libssl1.0-dev but 1.0.2o-1 is to be installed                                                                                          libssl1.0-dev : Conflicts: libssl-dev but 1.1.1-2 is to be installed                                                                                          E: Unable to correct problems, you have held broken packages.   
```

2. I got it to install anything from 2.1 to 2.5 with the "fix", but 1.9 is still a no go.
